src/index.ts

```diff
@@ -20,12 +34,53 @@ export const rules = {
 	"valid-version": validVersion,
 };
 
+const recommendedRuleSettings = Object.fromEntries(
+	Object.entries(rules)
+		.filter(([, rule]) => rule.meta.docs?.recommended)
+		.map(([name]) => ["package-json/" + name, "error" as const]),
+);
+
+interface Configs {
+	recommended: {
+		rules: typeof recommendedRuleSettings;
+	};
+	"recommended-flat": {
+		files: string[];
+		languageOptions: {
+			parser: typeof parserJsonc;
+		};
+		plugins: {
+			"package-json": typeof plugin;
+		};
+		rules: typeof recommendedRuleSettings;
+	};
+}
+
+const plugin = {
+	configs: {} as Configs,
+	meta: {
+		name,
+		version,
+	},
+	rules,
+};
+
 export const configs = {
 	recommended: {
-		rules: Object.fromEntries(
-			Object.entries(rules)
-				.filter(([, rule]) => rule.meta.docs?.recommended)
-				.map(([name]) => ["package-json/" + name, "error" as const]),
-		),
+		rules: recommendedRuleSettings,
+	},
+	["recommended-flat"]: {
```

Comment:
"Flat" isn't a great long-term way to refer to the configs. ESLint will be deprecating eslintrc configs soon, so all configs will be flat. "Flat" as a term will be redundant.

My preference would be to expose them both as `configs.recommended`. As in, only having one config property. The new version is essentially a superset of the old one (both have the same `rules`), and when I tried unifying them locally it all seemed nice and working. https://github.com/solidjs-community/eslint-plugin-solid takes this approach too and it seems to be working.

If that doesn't work _(and I'd really want to know - why not?)_ then I'd prefer calling the old config `recommended-legacy` or something like that the way [`eslint-plugin-markdown`](https://github.com/eslint/eslint-plugin-markdown/pull/232) does.

Reply:
This idea is great, I have refactored the code to use the same strategy as eslint-plugin-solid. I used `recommended-flat` before because using `recommended-legacy` may be a breaking change.